tests/unit_tests/wallet_storage.cpp

```diff
@@ -0,0 +1,229 @@
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "unit_tests_utils.h"
+#include "gtest/gtest.h"
+
+#include "file_io_utils.h"
+#include "wallet/wallet2.h"
+
+using namespace boost::filesystem;
+using namespace epee::file_io_utils;
+
+static constexpr const char WALLET_00fd416a_PRIMARY_ADDRESS[] =
+    "45p2SngJAPSJbqSiUvYfS3BfhEdxZmv8pDt25oW1LzxrZv9Uq6ARagiFViMGUE3gJk5VPWingCXVf1p2tyAy6SUeSHPhbve";
+
+TEST(wallet_storage, store_to_file2file)
+{
+    const path source_wallet_file = unit_test::data_dir / "wallet_00fd416a";
+    const path interm_wallet_file = unit_test::data_dir / "wallet_00fd416a_copy_file2file";
+    const path target_wallet_file = unit_test::data_dir / "wallet_00fd416a_new_file2file";
+
+    ASSERT_TRUE(is_file_exist(source_wallet_file.string()));
+    ASSERT_TRUE(is_file_exist(source_wallet_file.string() + ".keys"));
+
+    copy_file(source_wallet_file, interm_wallet_file, copy_option::overwrite_if_exists);
+    copy_file(source_wallet_file.string() + ".keys", interm_wallet_file.string() + ".keys", copy_option::overwrite_if_exists);
+
+    ASSERT_TRUE(is_file_exist(interm_wallet_file.string()));
+    ASSERT_TRUE(is_file_exist(interm_wallet_file.string() + ".keys"));
+
+    if (is_file_exist(target_wallet_file.string()))
+        remove(target_wallet_file);
+    if (is_file_exist(target_wallet_file.string() + ".keys"))
+        remove(target_wallet_file.string() + ".keys");
+    ASSERT_FALSE(is_file_exist(target_wallet_file.string()));
+    ASSERT_FALSE(is_file_exist(target_wallet_file.string() + ".keys"));
+
+    epee::wipeable_string password("beepbeep");
+    tools::wallet2 w;
+    w.load(interm_wallet_file.string(), password);
+    const std::string primary_address = w.get_address_as_str();
+    EXPECT_EQ(WALLET_00fd416a_PRIMARY_ADDRESS, primary_address);
+    w.store_to(target_wallet_file.string(), password);
+
+    EXPECT_FALSE(is_file_exist(interm_wallet_file.string()));
+    EXPECT_FALSE(is_file_exist(interm_wallet_file.string() + ".keys"));
+    EXPECT_TRUE(is_file_exist(target_wallet_file.string()));
+    EXPECT_TRUE(is_file_exist(target_wallet_file.string() + ".keys"));
```

Comment:
Testing closing + opening the new file would be nice here (untested on my end)

```suggestion
    const auto files_are_expected = [&]()
    {
        EXPECT_FALSE(is_file_exist(interm_wallet_file.string()));
        EXPECT_FALSE(is_file_exist(interm_wallet_file.string() + ".keys"));
        EXPECT_TRUE(is_file_exist(target_wallet_file.string()));
        EXPECT_TRUE(is_file_exist(target_wallet_file.string() + ".keys"));
    };

    {
        tools::wallet2 w;
        w.load(interm_wallet_file.string(), password);
        const std::string primary_address = w.get_address_as_str();
        EXPECT_EQ(WALLET_00fd416a_PRIMARY_ADDRESS, primary_address);
        w.store_to(target_wallet_file.string(), password);
        files_are_expected();
    }

    files_are_expected();

    {
        tools::wallet2 w;
        w.load(target_wallet_file.string(), password);
        const std::string primary_address = w.get_address_as_str();
        EXPECT_EQ(WALLET_00fd416a_PRIMARY_ADDRESS, primary_address);
        files_are_expected();
    }

    files_are_expected();
```